sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala

```diff
@@ -1523,8 +1535,30 @@ abstract class DynamicPartitionPruningSuiteBase
   }
 }
 
-class DynamicPartitionPruningSuiteAEOff extends DynamicPartitionPruningSuiteBase
+class DynamicPartitionPruningV1SuiteAEOff extends DynamicPartitionPruningSuiteBase
   with DisableAdaptiveExecutionSuite
 
-class DynamicPartitionPruningSuiteAEOn extends DynamicPartitionPruningSuiteBase
+class DynamicPartitionPruningV1SuiteAEOn extends DynamicPartitionPruningSuiteBase
   with EnableAdaptiveExecutionSuite
+
+class DynamicPartitionPruningV2SuiteAEOff extends DynamicPartitionPruningSuiteBase
```

Comment:
Do you think we can spin off this test case expanding contributions? It looks like we can merge this independently first.